### PR TITLE
(PC-25445) feat(SearchLocationModal): disconnect radius geoloc loc

### PR DESCRIPTION
**Pull request by NolwenMajorFrances**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-25445

## Checklist

I have:

- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [x] Written **unit tests** native (and web when implementation is different) for my feature.
- [x] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change

## Screenshots

https://github.com/pass-culture/pass-culture-app-native/assets/144016890/10df9bac-447c-470e-8d4a-b0658411082e

### CODE REVIEW

File: src/features/location/components/SearchLocationModal.tsx [hunk 1]
```diff
@@ -136,18 +141,27 @@ export const SearchLocationModal = ({
...
   }
 
-  const onValuesChange = useCallback(
+  const onAroundRadiusPlaceValueChange = useCallback(
+    (newValues: number[]) => {
+      if (visible) {
+        setAroundRadiusPlace(newValues[0])
+      }
+    },
+    [visible]
```
Comment (TristanMaurier98) [inline src/features/location/components/SearchLocationModal.tsx:156]: tu ne devrais pas avoir setAroundMeRadius en dépendance ici ?

File: src/features/location/components/SearchLocationModal.tsx [hunk 2]
```diff
@@ -136,18 +141,27 @@ export const SearchLocationModal = ({
...
+  )
+  const onAroundMeRadiusValueChange = useCallback(
     (newValues: number[]) => {
       if (visible) {
-        setAroundRadius(newValues[0])
+        setAroundMeRadius(newValues[0])
       }
     },
-    [setAroundRadius, visible]
+    [visible]
```
Comment (TristanMaurier98) [inline src/features/location/components/SearchLocationModal.tsx:164]: pareille ici (setAroundMeRadius)

File: src/features/location/components/SearchLocationModal.native.test.tsx [hunk 3]
```diff
@@ -107,6 +136,7 @@ describe('SearchLocationModal', () => {
   })
 
   it('should show geolocation modal if geolocation is never_ask_again on closing the modal after a geolocation button press', async () => {
+    const hideModalMock = jest.fn()
```
Comment (TristanMaurier98) [inline src/features/location/components/SearchLocationModal.native.test.tsx:139]: pourquoi ?

File: src/features/location/components/SearchLocationModal.native.test.tsx [hunk 4]
```diff
@@ -140,12 +170,171 @@ describe('SearchLocationModal', () => {
 
     expect(screen.getByText('Paramètres de localisation')).toBeOnTheScreen()
   })
+
+  describe('PlaceRadius', () => {
+    it("should equal default if wasn't set previously", async () => {
```
Comment (TristanMaurier98) [inline src/features/location/components/SearchLocationModal.native.test.tsx:175]: it wasn't je pense

File: src/features/location/components/SearchLocationModal.native.test.tsx [hunk 5]
```diff
@@ -140,12 +170,171 @@ describe('SearchLocationModal', () => {
...
+
+      const suggestedPlace = await screen.findByText(mockPlaces[0].label)
+      fireEvent.press(suggestedPlace)
+
+      expect(screen.getByText(radiusWithKm(DEFAULT_RADIUS))).toBeOnTheScreen()
+    })
+  })
+
+  describe('AroundMeRadius', () => {
+    it("should equal default if wasn't set previously", async () => {
```
Comment (TristanMaurier98) [inline src/features/location/components/SearchLocationModal.native.test.tsx:249]: pareille "it wasn't"

**Review by pierrecanthelou — APPROVED** [review]
RAS

**Pull request merged**